Client/Frontend/Browser/TemporaryDocument.swift

```diff
@@ -47,14 +44,52 @@ class TemporaryDocument: NSObject {
 
         let result = Deferred<URL>()
         pendingResult = result
-
-        downloadTask = session?.downloadTask(with: request)
-        downloadTask?.resume()
-
-        UIApplication.shared.isNetworkActivityIndicatorVisible = true
+        
+        if let tab = self.tab, let url = request.url {
+            ResourceDownloadManager.downloadResource(for: tab, url: url)
+            
+            ensureMainThread {
+                UIApplication.shared.isNetworkActivityIndicatorVisible = true
+            }
+        } else {
+            onDocumentDownloaded(document: nil, error: nil)
+        }
 
         return result
     }
+    
+    func onDocumentDownloaded(document: DownloadedResourceResponse?, error: Error?) {
+        ensureMainThread {
+            UIApplication.shared.isNetworkActivityIndicatorVisible = false
+        }
+        
+        // Store the blob/data in a local temporary file.
+        if let document = document, let data = document.data, !data.isEmpty {
+            let tempDirectory = URL(fileURLWithPath: NSTemporaryDirectory()).appendingPathComponent("TempDocs")
+            let url = tempDirectory.appendingPathComponent(filename)
+            
+            do {
+                try FileManager.default.createDirectory(at: tempDirectory, withIntermediateDirectories: true, attributes: nil)
```

Comment:
perhaps use try? if not catching any error?

Reply:
I need the error to fall through so I don't repeat code in the catch block and so the "promise" doesn't get fulfilled with the download url.

Reply:
Oh yeah :)